lib/api/machine/AdditionalSdmOptions.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright © 2019 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface AdditionalSdmConfiguration<ConfigurationType> {
```

Comment:
Doesn't an equivalent type already exist?

Reply:
I was thinking of this https://github.com/atomist/sdm/blob/e9e74b05c46730cb1d11ee116ac2c27f767684fd/lib/api/machine/SoftwareDeliveryMachineOptions.ts#L126 which seems different but related maybe.

Reply:
Where does `ConfigurationType` come from?

Reply:
yes, it is just a type parameter. Christian specifically suggested a wrapper type to put it inside the sdm property

Reply:
I'm not too happy with `AdditionalSdmConfiguration`: all the other config and options types call it `SoftwareDeliveryMachine` instead of `Sdm`. But I also don't have any other suggestion.